### ISSUE TITLE
'Product Summary' in All Products block is not pulling in the short description of the product

### ISSUE THREAD

**Issue by jrick1229** [woocommerce/woocommerce-gutenberg-products-block#2913]
**Describe the bug**

When using the 'All Products' block, adding a 'Product Summary' will result in an empty `<div class="wc-block-components-product-summary"></div>` in the source of the page where it's been placed.

**To Reproduce**

1. Create a new page with an 'All Products' block
2. Add a 'Product Summary' item within the block: https://d.pr/i/2HPEut & https://d.pr/i/9NX5l3
3. Click 'Done', then 'Update' the page
4. Create a test product **or** insert a 'Product short description' on an already existing one: https://d.pr/i/xoCYDX
5. Navigate to the new All Products block page and see that the 'Product Summary'/short description is empty: https://d.pr/i/faDgEo

**Expected behavior**

Per the block description, I would expect the product's short description to output where the block is placed within the All Products block: https://d.pr/i/vjSVc2

**Screenshots**

_Just reposted from above.._

- https://d.pr/i/2HPEut
- https://d.pr/i/9NX5l3
- https://d.pr/i/xoCYDX
- https://d.pr/i/faDgEo
- https://d.pr/i/vjSVc2

**WordPress (please complete the following information):**

<details>

```
### WordPress Environment ###

WordPress address (URL): http://localhost:10000
Site address (URL): http://localhost:10000
WC Version: 4.3.1
REST API Version: ✔ 1.0.10
WC Blocks Version: ✔ 2.7.2
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 1.3.1
Log Directory Writable: ✔
WP Version: 5.4.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.16.0
PHP Version: 7.3.5
PHP Post Max Size: 1,000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
cURL Version: 7.54.0
LibreSSL/2.6.5

SUHOSIN Installed: –
MySQL Version: 8.0.16
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.3.1
WC Database Prefix: wp_
Total Database Size: 28.96MB
Database Data Size: 22.38MB
Database Index Size: 6.58MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.23MB + Index: 0.23MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.14MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.09MB + Index: 0.06MB + Engine InnoDB
wp_aelia_dismissed_messages: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_aelia_exchange_rates_history: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_ariadminer_connections: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_automatewoo_abandoned_carts: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_automatewoo_customer_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_customers: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_automatewoo_events: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_guest_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_guests: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_automatewoo_log_meta: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_automatewoo_logs: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_queue: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_queue_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_referral_advocate_keys: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_automatewoo_referral_invites: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_referrals: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_automatewoo_unsubscribes: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.19MB + Index: 0.25MB + Engine InnoDB
wp_eum_logs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_followup_coupon_logs: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_followup_coupons: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_customer_carts: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_customer_notes: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_customer_orders: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_followup_customers: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_followup_email_excludes: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_followup_email_logs: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_followup_email_orders: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_followup_email_tracking: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_followup_followup_history: Data: 0.06MB + Index: 0.05MB + Engine InnoDB
wp_followup_order_categories: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_order_items: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_followup_subscriber_lists: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_subscribers: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_subscribers_to_lists: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_itsec_distributed_storage: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_itsec_fingerprints: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_itsec_geolocation_cache: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_itsec_lockouts: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_itsec_logs: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_itsec_mutexes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_itsec_opaque_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_itsec_temp: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_itsec_user_groups: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_ms_snippets: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 10.02MB + Index: 0.19MB + Engine InnoDB
wp_postmeta: Data: 1.52MB + Index: 2.00MB + Engine InnoDB
wp_posts: Data: 0.17MB + Index: 0.09MB + Engine InnoDB
wp_rank_math_404_logs: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_rank_math_internal_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_rank_math_internal_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_rank_math_redirections: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_rank_math_redirections_cache: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_rank_math_sc_analytics: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_say_what_strings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_smush_dir_images: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_snippets: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_taxjar_record_queue: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.08MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_booking_relationships: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_bookings_availability: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availabilitymeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.05MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.05MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_points_rewards_user_points: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_points_rewards_user_points_log: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wcpv_commissions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcpv_per_product_shipping_rules: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcs_payment_retries: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_exported_csv_items: Data: 0.08MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_gc_activity: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_gc_cards: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_gzd_dhl_labelmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_gzd_dhl_labels: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_gzd_shipment_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_gzd_shipment_items: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_gzd_shipmentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_gzd_shipments: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_gzd_shipping_provider: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_gzd_shipping_providermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_per_product_shipping_rules: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_pickup_locations_geodata: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_woocommerce_shipping_table_rates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_square_customers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpda_logging: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpda_media: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpda_menus: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpda_project: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wpda_project_page: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wpda_project_table: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpda_publisher: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wpda_table_design: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpda_table_settings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpfm_backup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpml_mails: Data: 7.19MB + Index: 0.00MB + Engine InnoDB
wp_wpmm_subscribers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_yith_wcmbs_downloads_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_yoast_seo_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_seo_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

attachment: 10
aw_workflow: 12
bookable_resource: 6
elementor_library: 1
fgf_master_log: 1
fgf_rules: 1
follow_up_email: 3
page: 18
post: 2
product: 42
product_variation: 17
revision: 34
shipping_package: 1
shop_coupon: 10
shop_order: 199
shop_order_refund: 4
shop_subscription: 80
wc_booking: 7
wc_membership_plan: 1
wc_order_status: 7
woo_discount: 2
woo_discount_cart: 2
wp_block: 1
wpcf7_contact_form: 1
xlwcty_thankyou: 1

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (5) ###

AutomateWoo: by WooCommerce – 4.8.3 – Installed version not tested with active version of WooCommerce 4.3.1
WooCommerce Stripe Gateway: by WooCommerce – 4.4.0 – Installed version not tested with active version of WooCommerce 4.3.1
WooCommerce Subscriptions: by WooCommerce – 3.0.5 – Installed version not tested with active version of WooCommerce 4.3.1
WooCommerce: by Automattic – 4.3.1
WP Mail Logging: by MailPoet – 1.9.5

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: accommodation-booking (accommodation-booking)
booking (booking)
external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #6 - /shop-1/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.5.5 (update to version 2.5.8 is available)
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Subscriptions ###

WCS_DEBUG: ✔ No
Subscriptions Mode: ✔ Live
Subscriptions Live URL: http://localhost:10000
Subscription Statuses: wc-active: 26
wc-expired: 1
wc-on-hold: 44
wc-cancelled: 9

WooCommerce Account Connected: ❌ No
Custom Retry Rules: ✔ No
Custom Retry Rule Class: ✔ No
Custom Raw Retry Rule: ✔ No
Custom Retry Rule: ✔ No
Retries Migration Status: ✔ Completed
Report Cache Enabled: ✔ Yes
Cache Update Failures: ✔ 0 failure

### Store Setup ###

Country / State: United Kingdom (UK)

### Subscriptions by Payment Gateway ###

other: wc-on-hold: 36
wc-active: 15
wc-cancelled: 3
wc-cancelled: 1

Stripe: wc-on-hold: 8
wc-expired: 1
wc-active: 9
wc-cancelled: 4


### Payment Gateway Support ###

Stripe: products
refunds
tokenization
add_payment_method
subscriptions
subscription_cancellation
subscription_suspension
subscription_reactivation
subscription_amount_changes
subscription_date_changes
subscription_payment_method_change
subscription_payment_method_change_customer
subscription_payment_method_change_admin
multiple_subscriptions
pre-orders


### Action Scheduler ###

Complete: 269
Oldest: 2020-06-27 08:55:53 -0400
Newest: 2020-07-01 09:04:56 -0400

Pending: 32
Oldest: 2020-07-25 14:21:38 -0400
Newest: 2020-08-24 10:16:14 -0400

Canceled: 7
Oldest: 2020-07-24 08:40:01 -0400
Newest: 2020-07-24 12:49:43 -0400
```

</details>

**Desktop (please complete the following information):**

* OS: macOS Catalina
* Browser: Firefox, Brave
* Version: v78.0.2, v1.11.97

**Additional context**

22895782-hc _(2020-07-25 17:32:33-2020-07-25 18:05:30)_
3172875-zen

**Comment by jrick1229** [comment]
I also just uploaded v3.0 to my site and still see the same issue.

**Comment by nerrad** [comment]
Thanks for the report! I can reproduce this in the 2.7 branch as well. @mikejolley this is likely a fix we'll want to get out for the Woo Core 4.5 release right? If so, that means we'll need a 3.0.1 release of blocks.

**Comment by nerrad** [comment]
Just to add some further clarification:

- This is reproducible on all release branches up to and including 3.0.0.
- It's related to #2607 adding Product Context Provider and Inner Product element blocks reading product data from the context provider. 
- This is not reproducible in the main branch because #2871 adds the fix for the Product Element blocks also used with All Products. #2817 was merged after 3.0 was branched.

None of the testing of branches from 2.7 - 3.0 caught this :(.  I'm currently working on what we're going to do here but we may just do an early 3.1.0 release.

**Comment by nerrad** [comment]
Just to confirm, this will be fixed by the 3.1.0 release (by code already in our main branch). So I'm going to close this.
